nautobot/extras/utils.py

```diff
@@ -207,6 +208,92 @@ def wrapper(model_class):
     return wrapper
 
 
+def update_registry_with_new_apps():
+    """
+    Updates the registry with new apps.
+
+    This function updates the registry model features for models.
+    """
+    from nautobot.extras.models import CustomField, Relationship, RelationshipAssociation, Role
+
+    lookup_confs = [
+        {
+            "model": CustomField,
+            "lookup_by": "field",
+            "field_names": ["_custom_field_data"],
+        },
+        {
+            "model": Relationship,
+            "lookup_by": "field",
+            "field_names": ["source_for_associations", "destination_for_associations"],
+            "extras": {"related_model": RelationshipAssociation},
+        },
+        {"model": Role, "lookup_by": "related_model", "related_model": Role},
+    ]
+
+    app_models = apps.get_models()
+    for lookup_conf in lookup_confs:
+        if lookup_conf["lookup_by"] == "field":
+            result = lookup_by_field(
+                app_models=app_models,
+                field_names=lookup_conf["field_names"],
+                extras=lookup_conf.get("extras"),
+            )
+        else:
+            result = lookup_by_related_model(app_models=app_models, related_model=lookup_conf["related_model"])
+        registry_items = {key: list(value) for key, value in result.items()}
```

Comment:
Can `lookup_by_field` and `lookup_by_related_model` just return the expected data structure directly, so that we don't have to transform `result` to `registry_items`?